### DISCUSSION TITLE
xtremestream.co not showing downloads

### DISCUSSION THREAD

**Discussion by jag176** [aclap-dev/video-downloadhelper#1473] [Help and Q&A]
### Your question

https://tube.perverzija.com/sexmex-katrina-moreno-5-de-mayo-fucking-my-mexican-hill-billy-sister/
as an example. Shows no media to process

### Addon & CoApp version and details

version: 9.1.1.0
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36
{
    "used_history_button": true,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "sort_by_status": false,
        "sort_reverse": true
    },
    "use_wide_ui": false,
    "use_legacy_ui": false,
    "open_count_store": 1,
    "successfull_dl": 1,
    "record_download_history": true,
    "history_limit_in_days": 31
}

**Comment by Mattmjs** [comment]
Yes it doesn't detect it since it's not in the proper format with the .m3u8 extension. See my screenshot in the developer tools below. 
The first one underlined is the master manifest for all the stream qualities. The next 480p and 720p, there are 1080p and 4k further down the list. I tried to use the master manifest adding the .m3u8 extension in ffmpeg but it returned the following

`[https @ 00000250fa913580] HTTP error 500 Internal Server Error
https://perverzija.xtremestream.xyz/player/xs1.php?data=03ea906731be85e38d49277fa858b7a1.m3u8: Server returned 5XX Server Error reply`

But adding  .m3u8 to the 480p version allowed ffmpeg to detect the stream. So as a workaround you could pick your quality and add
.m3u8 then paste it in a new tab and the extension should detect it.

![Screenshot 2024-11-18 184453](https://github.com/user-attachments/assets/7cf10243-c10d-47c7-ad07-f06fb4ac944b)

**Reply by jag176** [reply]
Ok, but I don't see that url on the original page or the extension. Is there something I'm supposed to click to get there?

**Reply by Mattmjs** [reply]
I mentioned I used the developer tools which is what you need to do. I forget to say that the links won't show until you play
the video first.

**Reply by jag176** [reply]
I'm using brave, I don't see those links in developer tools (I'm assuming you can use 'inspect element' to get to developer tools?) I just see this. It seems very difficult to find any download link
![element](https://github.com/user-attachments/assets/819103e7-e145-4fcb-9b33-41522cdd4500)

I even tried to do what you said manually with the example you sent me, it still doesn't download anything. 
https://perverzija.xtremestream.xyz/player/index.php?data=03ea906731be85e38d49277fa858b7a1@q=2160.m3u8

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user, like Matt.

This site uses totally non-standard methods of presenting its content.  I suspect either of these 2 hypotheses is correct:

1. The site developers are completely incompetent.

2. The site developers are very competent at making it nearly impossible for automatic downloading software (like VDH) to work.

Either way, I think it would be a lot to expect VDH to work on this site.  If I were the developer of VDH, I would look at this & say I'm not adapting my software to work with this totally rogue site.

Despite that, it is possible to download this content, as Matt has explained.  But you don't use inspect element to use your words.  I don't know Brave.  Brave is not supported by VDH anyway.  I am using Windows 11 Pro 64-bit version 24H2 build 26100.2314 & Firefox 132.0.2 64-bit.  In Firefox, you find the Developer Tools from a menu on the menu bar:

![#01](https://github.com/user-attachments/assets/14d46c43-bfd8-43c4-91c2-919155f4b670)

You should open a blank browser window & then activate the Developer Tools.  After that, open your porn web page in that browser window.  Then you must make the video play for a second or two.  Then you can pause it.  Then when you look in the Network Monitor (which is one of the components of the Developer Tools) you will see this:

![#02](https://github.com/user-attachments/assets/104c7c45-7635-4fca-bc7d-751e4cef08dd)

I like to sort my Network Monitor by the Type of the objects being displayed.  Just click the column heading to sort on that column.  In our case, we are looking for objects that are of type `plain`.  If you sort on Type, they will be grouped together.  This is the non-standard part.  HLS manifests are supposed to have file names with the extension m3u8.  These files are HLS manifests but they don't follow the proper naming convention.  In addition, HLS manifests are supposed to be of type `x-mpegurl` or `vnd.apple.mpegurl` or something very similar to those types.  They are not supposed to be of type `plain`.  Software like VDH would be expected to look for objects that obey the proper convention & be of type mpegurl or one of its mutations.  I would not expect VDH to handle such a non-standard web site as this.

In any case, I chose the HLS manifest highlighted in that image.  I got its URL using the popup context menu.  I corrected that URL by appending `.m3u8` to it.  Then I used ffmpeg to download the video:

[Porn mp4 Log.txt](https://github.com/user-attachments/files/17824894/Porn.mp4.Log.txt)

You can see in there that there is a high rate of error recovery going on.  I'm sure that contributed to the utterly putrid download speed I got, which is reported near the end of the log.  But the download did complete & I did have a video that played just fine in VLC.  I didn't actually watch it, just skimmed it to make sure I had video & audio right to the end.

So if you are willing to learn a bit about ffmpeg, you can get this content.  I wrote a tutorial on ffmpeg in our old Google Groups forum:

https://groups.google.com/g/video-downloadhelper-q-and-a/c/oYDFx367MSY

If you are motivated, you can read that & you'll learn how to do what I did to download this.
